### ISSUE TITLE
Nested Embedded Broken (postgres)

### ISSUE THREAD

**Issue by parisholley** [mikro-orm/mikro-orm#2362]
```
"stack": "SyntaxError: Unexpected token ')'
at new Function (<anonymous>)  
at Function.createFunction (node_modules/@mikro-orm/core/utils/Utils.js:612:52)  
at ObjectHydrator.getEntityHydrator (node_modules/@mikro-orm/core/hydration/ObjectHydrator.js:221:40)   
at ObjectHydrator.hydrate (node_modules/@mikro-orm/core/hydration/ObjectHydrator.js:23:30)
```

The snippet (cleansed) being generated is:

```js
 if (data.preferences != null) {
    entity.preferences = Object.create(prototype_preferences);
    if () {
      entity.preferences.review = Object.create(prototype_preferences_review);
    }
    if (data.preferences.review != null) {
      entity.preferences.review = Object.create(prototype_preferences_review);
    }
    if (data.preferences && typeof data.preferences.preference1 !== 'undefined') entity.preferences.preference1 = data.preferences.preference1;
}
```

```typescript
@Embeddable()
export class GrandChild {
  @Property()
  gc1 = false;

  @Property()
  gc2 = false;
}

@Embeddable()
export class Child {
  @Embedded(() => GrandChild)
  c1: GrandChild;
}

@Entity()
export class Parent {
  @Embedded(() => Child)
  child: Child;
}
```

as a note, i'm ripping these nested embeds out the project, but someone else may have use for it

```
    "@mikro-orm/cli": "^4.5.9",
    "@mikro-orm/core": "^4.5.9",
    "@mikro-orm/migrations": "^4.5.9",
    "@mikro-orm/nestjs": "^4.3.1",
    "@mikro-orm/postgresql": "^4.5.9",
    "@mikro-orm/reflection": "^4.5.9",
```

**Comment by B4nan** [comment]
Try changing order of discovery, I know multiple people had issues caused by order of embeddables during discovery. Nested embeddables are tested, including such use case, so if that don't help, full reproduction would be needed (ideally as a [single file test case](https://github.com/mikro-orm/mikro-orm/tree/master/tests/issues)). 

Also note that unless you use ts-morph, you need to specify types explicitly, type inference don't work with property initializers in reflect metadata:

```ts
  @Property()
  gc1: boolean = false;
```

Lastly let's not ignore issue templates, maintly the part with versions (I do understand you are probably using latest stable, but that can change over time).

**Comment by parisholley** [comment]
yep, using ts-morph. i'll revist with a test case once i get project coverted

**Comment by B4nan** [comment]
Fixed via b095d9e, also part of 4.x branch (so latest 4.x dev version).
